pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -228,6 +228,25 @@ stdenv.mkDerivation ({
         configureFlags+=" --extra-lib-dirs=$p/lib"
       fi
     done
+
+    if "${if stdenv.isDarwin then "true" else "false"}"; then
```

Comment:
It was mostly due to my lack of familiarity with Nix.  I've switched to @shlevy 's suggestion.